### PR TITLE
LPS-83811 Remove lodash from portal

### PR DESCRIPTION
**Pull request by iiventura**

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js [hunk 1]
```diff
@@ -192,7 +196,7 @@ Liferay = window.Liferay || {};
 
 				var config = args[1];
 
-				if (!_.isFunction(config) && !isNode(config)) {
+				if (!isFunction(config) && !isNode(config)) {
```
Comment (carloslancha) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js]: `Liferay.Util.isFunction`

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js [hunk 2]
```diff
@@ -222,15 +226,15 @@ Liferay = window.Liferay || {};
 		);
 
 		if (ioConfig.form) {
-			if (ioConfig.form.enctype == STR_MULTIPART && _.isFunction(window.FormData)) {
+			if (ioConfig.form.enctype == STR_MULTIPART && isFunction(window.FormData)) {
```
Comment (carloslancha) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js]: `Liferay.Util.isFunction`

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js [hunk 3]
```diff
@@ -304,7 +308,7 @@ Liferay = window.Liferay || {};
 		if (arguments.length === 1) {
 			var component = components[id];
 
-			if (component && _.isFunction(component)) {
+			if (component && isFunction(component)) {
```
Comment (carloslancha) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js]: `Liferay.Util.isFunction`

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/lodash/util.js [hunk 4]
```diff
@@ -5,16 +5,26 @@
 		{
 			bindKeyRight: function(context, key) {
 				var args = _.toArray(arguments).slice(2);
+				var bindKey=function () {
```
Comment (carloslancha) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/lodash/util.js]: SF and to `Liferay.Util.bindKey`

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/form_portlet.js [hunk 5]
```diff
@@ -792,15 +792,37 @@ AUI.add(
...
-							state1,
-							state2,
-							function(value1, value2, key) {
-								return (key === 'instanceId') || undefined;
-							}
-						);
+						return instance._isEqualExcept(state1,state2, "instanceId");
 					},
 
+					_isEqualExcept:function (object1, object2, propertyToIgnore) {
```
Comment (carloslancha) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/form_portlet.js]: to `Liferay.Util.isEqualExcept`

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/form_portlet.js [hunk 6]
```diff
@@ -792,15 +792,37 @@ AUI.add(
...
+							  }
+						  }
+
+						  return newObject;
+						}
+
+						object1 = removePropertyDeep(object1, propertyToIgnore);
+						object2 = removePropertyDeep(object2, propertyToIgnore);
+
+						return JSON.stringify(object1)===JSON.stringify(object2);
```
Comment (carloslancha) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/form_portlet.js]: SF

File: modules/apps/wiki/wiki-web/src/main/resources/META-INF/resources/wiki/js/WikiPortlet.es.js [hunk 7]
```diff
@@ -60,7 +60,18 @@ class WikiPortlet extends PortletBase {
 
 		let newFormat = formatSelect.options[formatSelect.selectedIndex].text.trim();
 
-		let confirmMessage = _.sub(
+		let sub= function(string,data){
```
[the remark this answers is not in the document]
Comment (iiventura) [reply modules/apps/wiki/wiki-web/src/main/resources/META-INF/resources/wiki/js/WikiPortlet.es.js]: Fixed

File: modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/miller_columns/LayoutColumn.es.js [hunk 8]
```diff
@@ -39,9 +39,18 @@ class LayoutColumn extends Component {
 			namespace: this.portletNamespace,
 			spritemap: this.pathThemeImages + '/lexicon/icons.svg'
 		};
+		var sub = function(string, data) {
```
[the remark this answers is not in the document]
Comment (iiventura) [reply modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/miller_columns/LayoutColumn.es.js]: Fixed for Liferay.Util.sub

File: modules/apps/asset/asset-publisher-web/src/main/resources/META-INF/resources/configuration/source.jsp [hunk 9]
```diff
@@ -458,11 +458,26 @@ List<AssetRendererFactory<?>> classTypesAssetRendererFactories = (List<AssetRend
 	);
 
 	function setDDMFields(className, name, value, displayValue, message) {
+		var escape=function(string) {
```
[the remark this answers is not in the document]
Comment (iiventura) [reply modules/apps/asset/asset-publisher-web/src/main/resources/META-INF/resources/configuration/source.jsp]: Fixed

File: modules/apps/asset/asset-tags-admin-web/src/main/resources/META-INF/resources/merge_tag.jsp [hunk 10]
```diff
@@ -132,7 +132,18 @@ renderResponse.setTitle(LanguageUtil.get(request, "merge-tags"));
 
 			tag = String(tag.html()).trim();
 
-			mergeText = _.sub(mergeText, mergeTagNamesArray, tag);
+			var sub=function(string,data) {
```
[the remark this answers is not in the document]
Comment (iiventura) [reply modules/apps/asset/asset-tags-admin-web/src/main/resources/META-INF/resources/merge_tag.jsp]: Fixed

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder_field_types_sidebar.js [hunk 11]
```diff
@@ -197,6 +197,31 @@ AUI.add(
 
 						var types = [];
 
+						var groupBy=function (collection,iteratee) {
```
[the remark this answers is not in the document]
Comment (iiventura) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder_field_types_sidebar.js]: Fixed

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder_field_types_sidebar.js [hunk 12]
```diff
@@ -197,6 +197,31 @@ AUI.add(
...
+							var finale=new Object();
+							var key;
+							//Verify if iteratee is a string or a function
```
[the remark this answers is not in the document]
Comment (iiventura) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder_field_types_sidebar.js]: Now that exist a Liferay.Util.groupBy I fixed ther

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder_sidebar.js [hunk 13]
```diff
@@ -230,6 +230,31 @@ AUI.add(
 
 						var types = [];
 
+						var groupBy= function groupBy(collection,iteratee) {
```
[the remark this answers is not in the document]
Comment (iiventura) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder_sidebar.js]: Fixed

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/lazy_load.js [hunk 14]
```diff
@@ -4,11 +4,15 @@
 		var modules;
 		var successCallback;
 
-		if (_.isArray(arguments[0])) {
+		var isFunction = function isFunction(fn) {
```
[the remark this answers is not in the document]
Comment (iiventura) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/lazy_load.js]: Fixed

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/lazy_load.js [hunk 15]
```diff
@@ -4,11 +4,15 @@
...
+			return (!!fn && (typeof fn == 'object' || typeof fn == 'function')) && toString.call(fn) == '[object Function]';
+		};
+
+		if (Array.isArray(arguments[0])) {
 			modules = arguments[0];
 
-			successCallback = _.isFunction(arguments[1]) ? arguments[1] : null;
-			failureCallback = _.isFunction(arguments[2]) ? arguments[2] : null;
+			successCallback = isFunction(arguments[1]) ? arguments[1] : null;
```
[the remark this answers is not in the document]
Comment (iiventura) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/lazy_load.js]: Fixed

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/lazy_load.js [hunk 16]
```diff
@@ -4,11 +4,15 @@
...
+			failureCallback = isFunction(arguments[2]) ? arguments[2] : null;
```
[the remark this answers is not in the document]
Comment (iiventura) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/lazy_load.js]: Fixed

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/lazy_load.js [hunk 17]
```diff
@@ -17,9 +21,9 @@
 				if (_.isString(arguments[i])) {
 					modules[i] = arguments[i];
 				}
-				else if (_.isFunction(arguments[i])) {
+				else if (isFunction(arguments[i])) {
```
[the remark this answers is not in the document]
Comment (iiventura) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/lazy_load.js]: Fixed

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/lazy_load.js [hunk 18]
```diff
@@ -17,9 +21,9 @@
...
 					successCallback = arguments[i];
-					failureCallback = _.isFunction(arguments[++i]) ? arguments[i] : null;
+					failureCallback = isFunction(arguments[++i]) ? arguments[i] : null;
```
[the remark this answers is not in the document]
Comment (iiventura) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/lazy_load.js]: Fixed

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js [hunk 19]
```diff
@@ -5,6 +5,10 @@ Liferay = window.Liferay || {};
 		return node && (node._node || node.jquery || node.nodeType);
 	};
 
+	var isFunction = function(fn) {
```
[the remark this answers is not in the document]
Comment (iiventura) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js]: Fixed

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js (hunk 2 above)
Comment (iiventura) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js]: Fixed

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js (hunk 3 above)
Comment (iiventura) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js]: Fixed

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/form_portlet.js (hunk 5 above)
Comment (iiventura) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/form_portlet.js]: Fixed

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/form_portlet.js (hunk 6 above)
Comment (iiventura) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/form_portlet.js]: Fixed

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/lodash/util.js (hunk 4 above)
Comment (iiventura) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/lodash/util.js]: Fixed

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/util.js [hunk 20]
```diff
@@ -81,7 +81,24 @@ AUI.add(
...
+					var propName = aProps[i];
+
+					// If values of same property are not equal,
+					// objects are not equivalent
+					if (a[propName] !== b[propName]) {
+						return false;
+					}
+				}
+
+				return true;
```
Comment (iiventura) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/util.js]: Replace this  function for Liferay.Util.isEqual

**Pull request closed without merging**